### DISCUSSION TITLE
Best Practices for HTML to be posted on-chain?

### DISCUSSION THREAD

**Discussion by Keeeeeeeks** [WalletConnect/walletconnect-monorepo#3038] [Developer Support]
Hi all,

I'm working on a project that will be posted on IPFS/Arweave, and have web3modal for connecting to wallet, signing transactions, and signing messages.

What are the best practices for protecting secrets in that case?

Am I able to import web3modal logic in the project itself?
Or would it be better to post the project in a way that passes through or inherits an injectedprovider from another site, and make it so that the project is viewed from said other site via iframe?

**Comment by chris13524** [comment]
I assume you are referring to protecting the project ID?

Project ID is expected to be public, as it is on any other frontend that uses Web3Modal. So it's fine to publish the project ID to IPFS.
